### PR TITLE
RYA-284: updated to support bloom filter

### PR DESCRIPTION
**Pull request by thn-w**
## Description
Updated dao/accumulo.rya to support the use of bloom filter

### Tests
no new tests

### Links
[JIRA -> RYA-284](https://issues.apache.org/jira/browse/RYA-284)

### People To Review
@amihalik

### CODE REVIEW

**Comment by asfgit** [comment]
Can one of the admins verify this patch?

**Comment by amihalik** [comment]
Can you submit another PR with only the required changes?  There are a lot of whitespace and final changes.

Normally I would power through this, but the github "ignore whitespace" hack doesn't seem to be working.

File: dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java [hunk 1]
```diff
@@ -374,56 +387,64 @@ protected String getData(RyaType ryaType) {
...
+        }
+        catch (final Exception e) {
             throw new RyaDAOException(e);
         }
     }
 
-    protected void fillScanner(ScannerBase scanner, RyaURI context, String qualifier, Long ttl, Long currentTime,
-            TripleRowRegex tripleRowRegex, RdfCloudTripleStoreConfiguration conf) throws IOException {
-        if (context != null && qualifier != null) {
+    private static Text createText(final byte[] data, final RyaType ryaType) throws RyaTypeResolverException {
```
Comment (amihalik) [inline dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java]: "appendType" would be a better name for this method.

File: dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java [hunk 2]
```diff
@@ -304,67 +311,73 @@ protected String getData(RyaType ryaType) {
...
 
                 // use range to set scanner
                 // populate scanner based on authorizations, ttl
                 layout = entry.getKey();
-                ByteRange byteRange = entry.getValue();
+                final ByteRange byteRange = entry.getValue();
+
                 Range range = new Range(new Text(byteRange.getStart()), new Text(byteRange.getEnd()));
+                if ((stmt.getSubject() != null) && (stmt.getPredicate() != null) && (stmt.getObject() != null)) {
+                    range = Range.exact(createText(byteRange.getStart(), stmt.getObject()));
```
Comment (amihalik) [inline dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java]: does `byteRange.getStart()` ever have the type appended to it at this point?  (if so, then this won't work)

File: dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java (hunk 1 above)
Comment (thn-w) [inline dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java]: Okay. Will put up a new PR with the new method name.

File: dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java (hunk 2 above)
Comment (thn-w) [inline dao/accumulo.rya/src/main/java/org/apache/rya/accumulo/query/AccumuloRyaQueryEngine.java]: that what createText (current method name) is supposed to do.

**Comment by thn-w** [comment]
Closed and will submit a new PR as requested.

**Pull request closed without merging**